### PR TITLE
Fix all issues with ESLint

### PR DESCRIPTION
**Pull request by w3bdesign**
Fixed all errors from ESLint

### CODE REVIEW

**Comment by w3bdesign** [comment]
@acelaya

**Comment by w3bdesign** [comment]
All checks have passed
11 successful checks

File: rollup.config.js [hunk 1]
```diff
@@ -7,12 +8,12 @@ export default [
...
-        typescript: require('typescript'),
-      }),
+        typescript: require('typescript')
+      })
     ],
     output: [
       { file: main, format: 'cjs' },
-      { file: module, format: 'esm' },
-    ],
-  },
```
Comment (acelaya) [inline rollup.config.js:17]: Please, keep all trailing commas you removed.

File: src/ExternalLink.tsx [hunk 2]
```diff
@@ -1,15 +1,15 @@
-import React, { FunctionComponent, ReactNode } from 'react';
```
Comment (acelaya) [inline src/ExternalLink.tsx:1]: I prefer explicit imports. Don't think it's necessary to change this.

File: src/ExternalLink.tsx [hunk 3]
```diff
@@ -1,15 +1,15 @@
...
 
 export interface ExternalLinkProps {
   href: string;
-  children?: ReactNode;
+  children?: React.ReactNode;
   [rest: string]: any;
 }
 
-const ExternalLink: FunctionComponent<ExternalLinkProps> = ({ href, children, ...rest }: ExternalLinkProps) => (
+const externalLink: React.FunctionComponent<ExternalLinkProps> = ({ href, children, ...rest }: ExternalLinkProps) => (
```
Comment (acelaya) [inline src/ExternalLink.tsx:9]: Why do you change the first character to lowercase? It's a breaking change and has nothing to do with linting.

Also, react components usually start with uppercase by convention.

File: test/ExternalLink.spec.tsx [hunk 4]
```diff
@@ -4,7 +4,7 @@ import '@testing-library/jest-dom/extend-expect';
 import ExternalLink from '../src/ExternalLink';
 
 describe('<ExternalLink />', () => {
-  const createExternalLink = (href: string = 'href', children?: ReactNode): ChildNode => {
```
Comment (acelaya) [inline test/ExternalLink.spec.tsx:7]: No need to remove this. Lets keep it as it was.

**Review by acelaya — CHANGES_REQUESTED** [review]

**Comment by acelaya** [comment]
Please, keep the scope of the PR as reduced add possible. The package lock has 16k new lines, that makes no sense.

Please, roll that change back

File: src/ExternalLink.tsx (hunk 3 above)
Comment (w3bdesign) [reply src/ExternalLink.tsx:9]: Eslint was complaining about this.

File: src/ExternalLink.tsx (hunk 2 above)
Comment (w3bdesign) [reply src/ExternalLink.tsx:1]: Eslint would complain before I changed this.

File: rollup.config.js (hunk 1 above)
Comment (w3bdesign) [reply rollup.config.js:17]: Eslint was complaining both when I had a trailing comma and without, which was why I changed it.

File: package.json [hunk 5]
```diff
@@ -35,7 +35,7 @@
     "@testing-library/jest-dom": "^5.11.6",
     "@testing-library/react": "^11.2.2",
     "@types/jest": "^26.0.19",
-    "@types/react": "^17.0.0",
+    "@types/react": "^17.0.1",
```
[the remark this answers is not in the document]
Comment (w3bdesign) [reply package.json:38]: I believe this fixed the missing type declaration erroe I got

File: .eslintrc [hunk 6]
```diff
@@ -16,7 +17,8 @@
     }
   },
   "rules": {
-    "comma-dangle": ["error", "always-multiline"],
+    "comma-dangle": ["warn", "always-multiline"],
+    "@typescript-eslint/no-unsafe-assignment": "warn",
```
[the remark this answers is not in the document]
Comment (w3bdesign) [reply .eslintrc:21]: Eslint was complaing borh with and without trailing commas

File: .eslintrc [hunk 7]
```diff
@@ -16,7 +17,8 @@
     }
   },
   "rules": {
-    "comma-dangle": ["error", "always-multiline"],
+    "comma-dangle": ["warn", "always-multiline"],
```
[the remark this answers is not in the document]
Comment (w3bdesign) [reply .eslintrc:20]: Eslinth was complaining both with it on and off

File: test/ExternalLink.spec.tsx (hunk 4 above)
Comment (w3bdesign) [reply test/ExternalLink.spec.tsx:7]: I think npx eslint --fix removed it

**Pull request closed without merging**